backend/dataall/db/api/share_object.py

```diff
@@ -309,28 +604,61 @@ def reject_share_object(
     ) -> models.ShareObject:
 
         share = ShareObject.get_share_by_uri(session, uri)
+        dataset = api.Dataset.get_dataset_by_uri(session, share.datasetUri)
+        share_items_states = ShareObject.get_share_items_states(session, uri)
+
+        Share_SM = ShareObjectSM(share.status)
+        new_share_state = Share_SM.run_transition(ShareObjectActions.Reject.value)
+
+        for item_state in share_items_states:
+            Item_SM = ShareItemSM(item_state)
+            new_state = Item_SM.run_transition(ShareObjectActions.Reject.value)
+            Item_SM.update_state(session, share.shareUri, new_state)
+
+        Share_SM.update_state(session, share, new_share_state)
+
+        ResourcePolicy.delete_resource_policy(
+            session=session,
+            group=share.groupUri,
+            resource_uri=dataset.datasetUri,
+        )
+        api.Notification.notify_share_object_approval(session, username, dataset, share)
```

Comment:
Think this should be calling `notify_share_object_rejection` rather than `notify_share_object_approval`

Reply:
You are right, let me review the notifications. I have also noticed that in general the notifications can be improved. They do not show the principal of the share just the dataset that has been approved. I will open another issue for this.

Reply:
https://github.com/awslabs/aws-dataall/issues/291 Here is the issue for notifications